pkg/vault/contracts/Authorizer.sol

```diff
@@ -32,258 +32,139 @@ import "./interfaces/IAuthorizer.sol";
-     * you perform all queries on the same block. See the following
-     * https://forum.openzeppelin.com/t/iterating-over-elements-on-enumerableset-in-openzeppelin-contracts/2296
-     * for more information.
-     */
-    function getRoleGlobalMember(bytes32 role, uint256 index) public view returns (address) {
-        return _roles[role].globalMembers.at(index);
-    }
-
-    /**
-     * @dev Returns the number of accounts that have `role` as global permission. Can be used
-     * together with {getRoleGlobalMember} to enumerate all bearers of a role.
-     */
-    function getRoleMemberCountByContract(bytes32 role, address where) public view returns (uint256) {
-        return _roles[role].membersByContract[where].length();
-    }
-
-    /**
-     * @dev Returns one of the accounts that have `role` in contract `where`. `index` must be a
-     * value between 0 and {getRoleMemberCountByContract}, non-inclusive.
-     *
-     * Role bearers are not sorted in any particular way, and their ordering may
-     * change at any point.
-     *
-     * WARNING: When using {getRoleMemberByContract} and {getRoleMemberCountByContract}, make sure
-     * you perform all queries on the same block. See the following
-     * https://forum.openzeppelin.com/t/iterating-over-elements-on-enumerableset-in-openzeppelin-contracts/2296
-     * for more information.
-     */
-    function getRoleMemberByContract(
-        bytes32 role,
-        uint256 index,
+    function canPerform(
+        bytes32 action,
+        address account,
         address where
-    ) public view returns (address) {
-        return _roles[role].membersByContract[where].at(index);
-    }
-
-    /**
-     * @dev Returns the admin role that controls `role`. See {grantRole} and
-     * {revokeRole}.
-     *
-     * To change a role's admin, use {_setRoleAdmin}.
-     */
-    function getRoleAdmin(bytes32 role) public view returns (bytes32) {
-        return _roles[role].adminRole;
+    ) public view override returns (bool) {
+        return isAllowed[permissionId(action, account, where)] || isAllowed[permissionId(action, account, ANYWHERE)];
     }
 
     /**
-     * @dev Grants multiple roles to a single account for a set of contracts.
+     * @dev Grants multiple permissions to a single account.
      */
-    function grantRoles(
-        bytes32[] memory roles,
+    function grantPermissions(
+        bytes32[] memory actions,
         address account,
-        address[] calldata where
+        address[] memory where
     ) external {
-        _require(where.length > 0, Errors.INPUT_LENGTH_MISMATCH);
-        for (uint256 i = 0; i < roles.length; i++) {
-            _require(isAdmin(roles[i], msg.sender), Errors.GRANT_SENDER_NOT_ADMIN);
+        for (uint256 i = 0; i < actions.length; i++) {
             for (uint256 j = 0; j < where.length; j++) {
-                _grantRole(roles[i], account, where[j]);
+                grantPermission(actions[i], account, where[j]);
```

Comment:
I'm generally not a big fan of calling public non-view functions from other functions, as you need to be very careful about `msg.sender`. This case is simple enough that it can be fine.

However I'd keep things where we left them in #1109, and just keep the array functions.